### DISCUSSION TITLE
when i call update api, i get this error: unknown argument 'id'. available options are listed in green.

### DISCUSSION THREAD

**Discussion by icoomn** [prisma/prisma#20427] [Q&A] [answered]
```js
// this is the error message
update(id: number, updateArticleDto: UpdateArticleDto) {
     return this.prisma.article.update({
           where: {
             id: 2
           },
           data: {
             id: 2,
             ~~
             title: "正则入门推荐及插件分享，好用的很！",
             coverImg: "图片",
             intro: "简介",
             content: "文章内容",
             viewCount: 0,
             status: 0,
             createTime: "2023-07-25T03:26:37.118Z",
             updateTime: "2023-07-25T03:26:37.118Z",
             categoryId: 1,
             commentCount: 0,
             categoryName: "HTML/CSS"
           }
         })

Unknown argument `id`. Available options are listed in green.
PrismaClientValidationError: 
Invalid `this.prisma.article.update()` invocation in
```

**Comment by ludralph** [comment, marked as answer]
Hi @oeemn :wave: 

This error is thrown because you are trying to update the `id` field in the data object, which is not allowed. The `id` field is usually used as a unique identifier for each row in the database, and it should not be updated directly.

To update an existing article using Prisma, you only need to specify the where clause to identify which article you want to update, and then provide the fields and their new values in the data object. 

Can you share the complete error log and schema for your project to understand better the options listed in green and allowable fields for the `article` model.

**Reply by icoomn** [reply]
Thank you for your help.  you are right, the 'id' can not be updated. 

but the real reason is not that, cause prisma `UpdateArticleDto` have already help us remove the 'id'.

```js
export class UpdateArticleDto extends PartialType(CreateArticleDto) {}
```


this is my schema

```prisma
model Article {
  id         Int      @id @default(autoincrement())
  title      String   @default("")
  coverImg   String   @default("")
  intro      String   @default("")
  content    String   @default("")
  viewCount  Int      @default(0)
  status     Int      @default(1)
  createTime DateTime @default(now())
  updateTime DateTime @default(now()) @updatedAt()
  categoryId Int
}
```

this is my parameters
```js
             id: 2,
             title: "正则入门推荐及插件分享，好用的很！",
             coverImg: "图片",
             intro: "简介",
             content: "文章内容",
             viewCount: 0,
             status: 0,
             createTime: "2023-07-25T03:26:37.118Z",
             updateTime: "2023-07-25T03:26:37.118Z",
             categoryId: 1,
             commentCount: 0,
             categoryName: "HTML/CSS"
```

the error's reason is the `commentCount` and `categoryName`, this two fields are not in the schema (createArticleDto), so. when i call update api get this error. now i delete this two fields from the parameters, the error is clear.

**Comment by MDileg** [comment]
Hello everybody, I have a problem create a data, Please help me about that
  async create(dto: CreatePostDto) {
    const post = await this.prisma.post.create({
      data: {
        ...dto,
        preview_image: "",
        brands: {},
        categories: {},
        ratings: 5.0,
      },
    });
    return post;
  }


THIS IS MY DTO...
export class CreatePostDto {
  @IsString()
  name_tm: string;

  @IsString()
  name_ru: string;

  @IsBoolean()
  published: boolean;

  @IsNumber()
  price: number;

  @IsOptional()
  discount_price: number;

  @IsOptional()
  ratings: number;

  @IsString()
  description_tm: string;

  @IsString()
  description_ru: string;

  @IsBoolean()
  isPopular: boolean;

  @IsNumber()
  categoryId: never;

  @IsOptional()
  brandId: never;
}


OUTPUT => //Unknown argument `categoryId`. Did you mean `categories`? Available options are marked with ?.

**Reply by ludralph** [reply]
Hi @MDileg 👋 

Can you please create a new discussion as this one is closed?
